src/ripple/core/impl/SQLInterface_nodes.cpp

```diff
@@ -0,0 +1,1408 @@
+            R"sql(INSERT OR REPLACE INTO Ledgers
+                (LedgerHash,LedgerSeq,PrevHash,TotalCoins,ClosingTime,PrevClosingTime,
+                CloseTimeRes,CloseFlags,AccountSetHash,TransSetHash)
+            VALUES
+                (:ledgerHash,:ledgerSeq,:prevHash,:totalCoins,:closingTime,:prevClosingTime,
+                :closeTimeRes,:closeFlags,:accountSetHash,:transSetHash);)sql");
+
+        auto db(ldgDB.checkoutDb());
+
+        soci::transaction tr(*db);
+
+        auto const hash = to_string(ledger->info().hash);
+        auto const parentHash = to_string(ledger->info().parentHash);
+        auto const drops = to_string(ledger->info().drops);
+        auto const closeTime =
+            ledger->info().closeTime.time_since_epoch().count();
+        auto const parentCloseTime =
+            ledger->info().parentCloseTime.time_since_epoch().count();
+        auto const closeTimeResolution =
+            ledger->info().closeTimeResolution.count();
+        auto const closeFlags = ledger->info().closeFlags;
+        auto const accountHash = to_string(ledger->info().accountHash);
+        auto const txHash = to_string(ledger->info().txHash);
+
+        *db << addLedger, soci::use(hash), soci::use(seq),
+            soci::use(parentHash), soci::use(drops), soci::use(closeTime),
+            soci::use(parentCloseTime), soci::use(closeTimeResolution),
+            soci::use(closeFlags), soci::use(accountHash), soci::use(txHash);
+
+        tr.commit();
+    }
+
+    return true;
+}
+
+std::optional<SQLLedgerInfo>
+loadLedgerInfo(
+    soci::session& session,
+    std::string const& sqlSuffix,
+    beast::Journal j)
+{
+    SQLLedgerInfo info;
+    std::string const sql =
+        "SELECT "
+        "LedgerHash, PrevHash, AccountSetHash, TransSetHash, "
+        "TotalCoins,"
+        "ClosingTime, PrevClosingTime, CloseTimeRes, CloseFlags,"
+        "LedgerSeq from Ledgers " +
+        sqlSuffix + ";";
+
+    session << sql, soci::into(info.sLedgerHash), soci::into(info.sPrevHash),
+        soci::into(info.sAccountHash), soci::into(info.sTransHash),
+        soci::into(info.totDrops), soci::into(info.closingTime),
+        soci::into(info.prevClosingTime), soci::into(info.closeResolution),
+        soci::into(info.closeFlags), soci::into(info.ledgerSeq64);
+
+    if (!session.got_data())
+    {
+        auto stream = j.debug();
+        JLOG(stream) << "Ledger not found: " << sqlSuffix;
+        return std::optional<SQLLedgerInfo>();
+    }
+
+    return std::optional<SQLLedgerInfo>(info);
+}
+
+std::optional<SQLLedgerInfo>
+loadLedgerInfoByIndex(
+    soci::session& session,
+    LedgerIndex ledgerSeq,
+    beast::Journal j)
+{
+    std::ostringstream s;
+    s << "WHERE LedgerSeq = " << ledgerSeq;
+    return loadLedgerInfo(session, s.str(), j);
+}
+
+std::optional<SQLLedgerInfo>
+loadOldestLedgerInfo(
+    soci::session& session,
+    beast::Journal j)
+{
+    std::ostringstream s;
+    s << "order by LedgerSeq asc limit 1";
```

Comment:
The convention is to capitalize any reserved words in the query, such as "ORDER", "BY", "ASC" and "LIMIT". Let's make sure all of the reserved words are capitalized in all of the queries.

Reply:
Done.